rules/ast/ast.go

```diff
@@ -800,6 +876,53 @@ func (node *VectorArithExpr) Eval(timestamp clientmodel.Timestamp) Vector {
 	panic("Invalid vector arithmetic expression operands")
 }
 
+// resultMetric returns the metric for the given sample based on the vector
+// binary operation and the matching options.
+func (node *VectorArithExpr) resultMetric(ls, rs *Sample) clientmodel.COWMetric {
+	if len(node.matchBy) > 0 {
+		m := clientmodel.Metric{}
+		for _, ln := range node.matchBy {
+			m[ln] = ls.Metric.Metric[ln]
+		}
+		for _, ln := range node.includeLabels {
```

Comment:
I can't seem to bottom it out, it always seems to be empty per helpers.go lines 125-129, as I can't find where we're getting that from the parser.